### ISSUE TITLE
Strange behavior of "next" button when in Files mode

### ISSUE THREAD

**Issue by wrybread** [vanilla-music/vanilla#685]
I use Vanilla Music exclusively in Files mode, meaning I'm browsing the file structure directly. If I open a folder and click a track and then press the "next" button, I'd expect it to skip to the next track in the folder. Instead it plays the track again.

I guess the logic is that the "next" button is playing the next track in the queue. But I think it's reasonable to assume that if there's nothing next in the queue it should simply play the next track in the folder.

Note that when I'm playing a song from the Tracks view, pressing "next" behaves as expected and advances to the next song.

Or maybe there's a way to make the "next" button work differently?

As it is, it makes the Files mode mostly useless, except when manually queueing files.

**Comment by adrian-bl** [comment]
Thats's not how Vanilla Music works: the forward & back buttons do not depend on your file listing: they only interact with the queue (dotmenu -> show queue).

Clicking on an item in the file view enqueues a single file, so tapping 'forward' wont do much.

Use the 'Play all' row to put the whole folder in the queue.

**Comment by wrybread** [comment]
> Thats's not how Vanilla Music works: the forward & back buttons do not depend on your file listing: they only interact with the queue (dotmenu -> show queue).

I understand the logic, but that's a different behavior than every other MP3 player I've used, even ones that, like Vanilla Music, support queues. Other players automatically add the rest of the folder or album or playlist to the queue. I think that's a much more natural behavior and I don't see a good rationale for Vanilla Music to be different, unless I'm missing something.

> Use the 'Play all' row to put the whole folder in the queue.

What if the user wants to play track 6 of an album (folder) to the end?

For whatever it's worth, I think a much more natural behavior is to automatically enqueue the rest of the folder when file is played. I suppose there could be an option to disable that behavior in settings.

**Comment by adrian-bl** [comment]
> Other players automatically add the rest of the folder or album or playlist to the queue. 

We do this for other tabs (and there is a setting to change the default behaviour) - but AFAIK this doesn't apply to the files tab, as we do not have a per-file 'Play All' action (need to double check)

So implementing the 'Play All' option for individual files would fix this use case (eg: hitting track 6 in a folder would play track 6 and enqueue all other items)

**Comment by wrybread** [comment]
> So implementing the 'Play All' option for individual files would fix this use case (eg: hitting track 6 in a folder would play track 6 and enqueue all other items)

Agreed, that would be a good fix. Especially if it were possible to do "play all" without extra presses, or long presses (in other words, making it the default action for play when in Files tab).
